### PR TITLE
[bugfix] normalizes uri configuration key names

### LINKED ISSUE #81: normalize uri configuration key names
> for context, see https://github.com/CatalystCode/3dtoolkit/pull/63#issuecomment-323124008

we should rename `turnServer.provider` config value to `turnServer.providerUri` and `authentication.authority` to `authentication.authorityUri` to be more consistent and clear that these are uri values.

This includes updating the documentation in the wiki, the config stubs, and the parsing logic.

### PR DESCRIPTION
**Pull request by KanishkT123**
Changed all the names to reflect whether or not they are URI's. Specifically changed the following in structs.h and everywhere else that is relevant:

1. `TurnServer.provider -> TurnServer.provider_uri`
2. `Authentication.authority -> Authentication.authority_uri`
3. `WebRTCConfig.server -> WebRTCConfig.server_uri`  

fixes #81

### CODE REVIEW

**Comment by bengreenier** [comment]
fixes #81

**Comment by bengreenier** [comment]
When we merge this we'll also need to update [the wiki](https://github.com/CatalystCode/3dtoolkit/wiki/JSON-Config-Files).

As this is technically a breaking change we'll want to land this is a major version bump release. As such, labeling as `invalid` for now (until we land on how to get that in).

**Comment by bengreenier** [comment]
Looks like this is branched off your work from #199 - we'll want to [rebase](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) this against `master` before we take the changes.

File: Libraries/ConfigParser/src/config_parser.cpp [hunk 1]
```diff
@@ -1,247 +1,247 @@
...
-		webrtcConfig->ice_configuration = root.get("iceConfiguration", NULL).asString();
-		if (root.isMember("turnServer"))
-		{
-			auto turnServerNode = root.get("turnServer", NULL);
-			if (turnServerNode.isMember("uri"))
-			{
-				webrtcConfig->turn_server.uri = turnServerNode.get("uri", NULL).asString();
-			}
-
-			if (turnServerNode.isMember("provider"))
```
Comment (bengreenier) [inline Libraries/ConfigParser/src/config_parser.cpp]: >if (turnServerNode.isMember("provider")) [](start = 3, length = 40)

do you think it makes sense to support both for a release or two? We should think through what the upgrade scenario is for a customer that currently has config using these 'old' values. 

The two ideas I have in my head right now are:
+ support old version temporarily
+ ship this with a major version bump to indicate breaking changes

Comment (KanishkT123) [reply Libraries/ConfigParser/src/config_parser.cpp]: Let's support the old version for now, and make it clear that support will eventually go away. I'll open an issue about it after I make the commit.

File: Libraries/ConfigParser/src/config_parser.cpp [hunk 2]
```diff
@@ -1,247 +1,266 @@
...
+			{
+				webrtcConfig->turn_server.uri = turnServerNode.get("uri", NULL).asString();
+			}
+
+			if (turnServerNode.isMember("providerUri"))
+			{
+				webrtcConfig->turn_server.provider_uri = turnServerNode.get("providerUri", NULL).asString();
+			}
+
+			if (turnServerNode.isMember("provider"))
```
Comment (bengreenier) [inline Libraries/ConfigParser/src/config_parser.cpp]: the ordering here is such that if both values are in place old will override the new - is that desired?

File: Libraries/ConfigParser/ConfigParser.Tests/ConfigParserTests.cpp [hunk 3]
```diff
@@ -57,11 +57,11 @@ namespace ConfigParserTests
 			Assert::AreEqual("test:test:1234", injectedWebRTCInstance->turn_server.uri.c_str());
 			Assert::AreEqual("test", injectedWebRTCInstance->turn_server.username.c_str());
 			Assert::AreEqual("test", injectedWebRTCInstance->turn_server.password.c_str());
-			Assert::AreEqual("test", injectedWebRTCInstance->server.c_str());
+			Assert::AreEqual("test", injectedWebRTCInstance->server_uri.c_str());
```
Comment (bengreenier) [inline Libraries/ConfigParser/ConfigParser.Tests/ConfigParserTests.cpp]: can we add a new test that validates this fallback behavior?

File: Libraries/ConfigParser/src/config_parser.cpp (hunk 2 above)
Comment (KanishkT123) [reply Libraries/ConfigParser/src/config_parser.cpp]: I can fix that, it certainly seems that the newer version should get priority.

File: Libraries/ConfigParser/ConfigParser.Tests/ConfigParserTests.cpp [hunk 4]
```diff
@@ -6,6 +6,8 @@
 #include <Windows.h>
 
 #include "config_parser.h"
+#include<iostream>
```
Comment (bengreenier) [inline Libraries/ConfigParser/ConfigParser.Tests/ConfigParserTests.cpp]: nit: we don't seem to use anything from iostream anymore, can this be removed?

File: Libraries/ConfigParser/ConfigParser.Tests/ConfigParserTests.cpp (hunk 3 above)
Comment (KanishkT123) [reply Libraries/ConfigParser/ConfigParser.Tests/ConfigParserTests.cpp]: Done! I had to change some of the code in config_parser and remove the hardcoded stuff, but there are now functional unit tests included.

**Pull request merged**